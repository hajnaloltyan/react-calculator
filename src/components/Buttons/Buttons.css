.buttons {
  display: grid;
  grid-template-columns: repeat(4, 1fr);
  grid-template-rows: auto repeat(4, 1fr);
  gap: 2px;
  background-color: #ddd;
  border: 1px solid #ccc;
  padding: 5px;
  width: 35vh;
}

.btn-text {
  grid-column: span 1;
  grid-row: span 1;
  background-color: #eee;
  border: 1px solid #ccc;
  padding: 10px;
  text-align: center;
  font-weight: 500;
  font-size: 16px;
}

.btn-text:hover {
  background-color: rgb(174, 255, 228);
}

.btn-0 {
  grid-column: span 2;
  grid-row: span 1;
}

.orange {
  background-color: #ff7b23;
}
