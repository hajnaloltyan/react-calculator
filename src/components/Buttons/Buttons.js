import React from 'react';
import PropTypes from 'prop-types';
import './Buttons.css';

const Buttons = ({ onButtonClick }) => {
  const handleClick = (buttonName) => {
    onButtonClick(buttonName);
  };

  const buttonNames = ['AC', '+/-', '%', 'รท', '7', '8', '9', 'x', '4', '5', '6', '-', '1', '2', '3', '+', '0', '.', '='];

  const renderButton = (buttonName, index) => {
    let classNames = 'btn-text';
    if ((index + 1) % 4 === 0 || index === 18) {
      classNames += ' orange';
    } else if (index === 16) {
      classNames += ' btn-0';
    }

    return (
      <button
        key={buttonName}
        className={classNames}
        onClick={() => handleClick(buttonName)}
        type="button"
      >
        {buttonName}
      </button>
    );
  };

  return (
    <article className="buttons">
      {buttonNames.map(renderButton)}
    </article>
  );
};

Buttons.propTypes = {
  onButtonClick: PropTypes.func.isRequired,
};

export default Buttons;
