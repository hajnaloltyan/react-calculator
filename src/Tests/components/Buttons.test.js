// import react-testing methods
import renderer from 'react-test-renderer';
import { render, screen } from '@testing-library/react';
// the component to test
import calculate from '../../components/logic/calculate';
import Buttons from '../../components/Buttons/Buttons';

let calculatorData = {
  total: null,
  next: null,
  operation: null,
};
const handleButtonClick = (buttonName) => {
  const newCalculatorData = calculate(calculatorData, buttonName);
  calculatorData = newCalculatorData;
};

describe('Testing the Buttons component', () => {
  test('Test if the button component render correctly', () => {
    const tree = renderer.create(<Buttons onButtonClick={handleButtonClick} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('Check if the AC button exist', () => {
    render(<Buttons onButtonClick={handleButtonClick} />);
    const ButtonsContainer = screen.getByText('AC');
    expect(ButtonsContainer).toHaveTextContent('AC');
  });

  test('Check if the % button exist', () => {
    render(<Buttons onButtonClick={handleButtonClick} />);
    const ButtonsContainer = screen.getByText('%');
    expect(ButtonsContainer).toHaveTextContent('%');
  });
});
