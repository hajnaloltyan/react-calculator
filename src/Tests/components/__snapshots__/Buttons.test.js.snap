// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing the Buttons component Test if the button component render correctly 1`] = `
<article
  className="buttons"
>
  <button
    className="btn-text"
    onClick={[Function]}
    type="button"
  >
    AC
  </button>
  <button
    className="btn-text"
    onClick={[Function]}
    type="button"
  >
    +/-
  </button>
  <button
    className="btn-text"
    onClick={[Function]}
    type="button"
  >
    %
  </button>
  <button
    className="btn-text orange"
    onClick={[Function]}
    type="button"
  >
    รท
  </button>
  <button
    className="btn-text"
    onClick={[Function]}
    type="button"
  >
    7
  </button>
  <button
    className="btn-text"
    onClick={[Function]}
    type="button"
  >
    8
  </button>
  <button
    className="btn-text"
    onClick={[Function]}
    type="button"
  >
    9
  </button>
  <button
    className="btn-text orange"
    onClick={[Function]}
    type="button"
  >
    x
  </button>
  <button
    className="btn-text"
    onClick={[Function]}
    type="button"
  >
    4
  </button>
  <button
    className="btn-text"
    onClick={[Function]}
    type="button"
  >
    5
  </button>
  <button
    className="btn-text"
    onClick={[Function]}
    type="button"
  >
    6
  </button>
  <button
    className="btn-text orange"
    onClick={[Function]}
    type="button"
  >
    -
  </button>
  <button
    className="btn-text"
    onClick={[Function]}
    type="button"
  >
    1
  </button>
  <button
    className="btn-text"
    onClick={[Function]}
    type="button"
  >
    2
  </button>
  <button
    className="btn-text"
    onClick={[Function]}
    type="button"
  >
    3
  </button>
  <button
    className="btn-text orange"
    onClick={[Function]}
    type="button"
  >
    +
  </button>
  <button
    className="btn-text btn-0"
    onClick={[Function]}
    type="button"
  >
    0
  </button>
  <button
    className="btn-text"
    onClick={[Function]}
    type="button"
  >
    .
  </button>
  <button
    className="btn-text orange"
    onClick={[Function]}
    type="button"
  >
    =
  </button>
</article>
`;
